Pilot/dirac-pilot.py

```diff
@@ -28,23 +28,53 @@
 import sys
 import time
 
+try:
+    from cStringIO import StringIO
+except ImportError:
+    from io import StringIO
+
 ############################
 # python 2 -> 3 "hacks"
 try:
-    from Pilot.pilotTools import Logger, pythonPathCheck, PilotParams, getCommand
+    from Pilot.pilotTools import (
+        Logger,
+        RemoteLogger,
+        pythonPathCheck,
+        PilotParams,
+        getCommand,
+    )
 except ImportError:
-    from pilotTools import Logger, pythonPathCheck, PilotParams, getCommand
+    from pilotTools import (
+        Logger,
+        RemoteLogger,
+        pythonPathCheck,
+        PilotParams,
+        getCommand,
+    )
 ############################
 
 if __name__ == "__main__":
 
     pilotStartTime = int(time.time())
 
-    log = Logger("Pilot", debugFlag=True)
-
+    sys.stdout, oldstdout = StringIO(), sys.stdout
     pilotParams = PilotParams()
-    if pilotParams.debugFlag:
-        log.setDebug()
+    sys.stdout, buffer = oldstdout, sys.stdout
```

Comment:
No,
First `sys.stdout` is replaced with a string buffer, so `PilotParams` (unchanged) is now logging to that buffer.
When finished `sys.stdout `is set back to its normal value, and a `buffer` is set  (effectively) to `StringIO`.
Then we can dump the buffer to stdout (normal logger): 

`sys.stdout.write(bufContent)`.

(please note that those records have been collected by a standard pilot logger so they have all the timestamps etc.)

Under a condition that the remote logger is active we can additionally ship the buffer content abroad:

`log.buffer.write(bufContent)`

Reply:
Ah I see, sorry my bad. I would still write line 62 as:

```
sys.stdout, buffer = oldstdout, StringIO()
```
to be more clear

Reply:
I don't think it is a good idea. This would rebind the `buffer,` we need one already filled with stuff, when it pretended to be `stdout` ;-)

Reply:
> I don't think it is a good idea. This would rebind the `buffer,` we need one already filled with stuff, when it pretended to be `stdout` ;-)

A-ha OK I understand now.